iphone/Classes/GeolocationModule.m

```diff
@@ -502,35 +421,73 @@ - (BOOL)hasCompass
 
 GETTER_IMPL(BOOL, hasCompass, HasCompass);
 
-- (void)performGeo:(NSString *)direction address:(NSString *)address callback:(GeolocationCallback *)callback
-{
-  [[TiApp app] startNetwork];
-
-  id aguid = TI_APPLICATION_GUID;
-  id sid = [[TiApp app] sessionId];
-
-  NSMutableDictionary *params = [@{
-    @"d" : direction,
-    @"aguid" : aguid,
-    @"mid" : [TiUtils appIdentifier],
-    @"sid" : sid,
-    @"q" : address,
-  } mutableCopy];
-
-  NSString *countryCode = [[NSLocale currentLocale] objectForKey:NSLocaleCountryCode];
-  if (countryCode) {
-    [params setValue:countryCode forKey:@"c"];
-  }
-  [callback start:params];
-  RELEASE_TO_NIL(params);
-}
-
 - (JSValue *)reverseGeocoder:(double)latitude longitude:(double)longitude withCallback:(JSValue *)callback
 {
 #ifndef __clang_analyzer__ // Ignore static analyzer error here, memory will be released. See TIMOB-19444
   KrollPromise *promise = [[[KrollPromise alloc] initInContext:[self currentContext]] autorelease];
   ReverseGeoCallback *rcb = [[ReverseGeoCallback alloc] initWithCallback:callback andPromise:promise];
-  [self performGeo:@"r" address:[NSString stringWithFormat:@"%f,%f", latitude, longitude] callback:rcb];
+
+  CLGeocoder *geoCoder = [[[CLGeocoder alloc] init] autorelease];
+  CLLocation *clLocation = [[[CLLocation alloc] initWithLatitude:latitude longitude:longitude] autorelease];
+  [geoCoder reverseGeocodeLocation:clLocation
+                   preferredLocale:[NSLocale currentLocale]
+                 completionHandler:^(NSArray<CLPlacemark *> *_Nullable placemarks, NSError *_Nullable error) {
+                   if (error != nil) {
+                     [rcb requestError:error];
+                   } else {
+                     NSMutableDictionary *events = [TiUtils dictionaryWithCode:0 message:nil];
+
+                     NSMutableArray *places = [NSMutableArray array];
+                     for (CLPlacemark *placemark in placemarks) {
+                       NSMutableDictionary *dict = [NSMutableDictionary dictionary];
+                       NSMutableString *address = [NSMutableString string];
```

Comment:
I don't think we should be building this string up ourselves. `CLPlacemark` has a `postalAddress` property which is a `CNPostalAddress` - you can then format it via a `CNPostalAddressFormatter`: https://developer.apple.com/documentation/contacts/cnpostaladdressformatter?language=objc

Something like:
``` obj-c
NSString *address = [CNPostalAddressFormatter stringFromPostalAddress:placemark.postalAddress style:CNPostalAddressFormatterStyleMailingAddress];
```

Reply:
Thanks for pointing. Updated.